test/ecto/type/enum_test.exs

```diff
@@ -0,0 +1,98 @@
+defmodule Ecto.Type.EnumTest do
```

Comment:
I am thinking this test is a bit too high-level for now. It probably makes sense to unit test those through the Ecto.Type API. We also need tests for equal?.